crates/fluvio-cli/src/consume/mod.rs

```diff
@@ -347,6 +351,16 @@ impl ConsumeOpt {
             builder.disable_continuous(true);
         }
 
+        if let Some(end_offset) = self.end_offset {
+            if end_offset <= 0 {
+                eprintln!("Argument end-offset must be greater than zero")
+            } else if end_offset > offset {
```

Comment:
I think that we should fail if end_offset is less than initial offset 
```suggestion
            } else if end_offset < offset {
```